### DISCUSSION TITLE
How to escape `$` characters in `it.each` strings?

### DISCUSSION THREAD

**Discussion by chipit24** [vitest-dev/vitest#3230] [General]
I have the following test:

```ts
it.each([
  [{ amount: 100 }, "$100.00"],
  [{ amount: 1200 }, "$1,200.00"],
])("formats %o as %s", (num, expected) => {
  expect(format(num)).toBe(expected);
});
```

These tests pass but it outputs:
```
formats { amount: 100 } as undefined
formats { amount: 1200 } as undefined,200.00
```

How can I have Vitest **not** try and interpret any `$`s? This only seems to happen when using `it.each` with objects (which I know [is documented](https://vitest.dev/api/#test-each), but I don't see a way to escape these characters).

**Comment by ghiscoding** [comment]
Just a wild guess, have you tried to double the symbol like `$$`? It's usually how to get over it in other languages, so I'm just guessing here

**Reply by chipit24** [reply]
I did, since that's the first thing ChatGPT recommended, and it didn't fix it.

**Reply by ghiscoding** [reply]
what about `$\$`
